internal/storage/bucket_handle.go

```diff
@@ -516,9 +516,20 @@ func (b *bucketHandle) GetFolder(ctx context.Context, folderName string) (*gcs.F
 	return folderResponse, err
 }
 
-func (b *bucketHandle) CreateFolder(ctx context.Context, folderName string) (folder *gcs.Folder, err error) {
-	// TODO: Implement me
-	return
+func (b *bucketHandle) CreateFolder(ctx context.Context, folderName string) (*gcs.Folder, error) {
+	req := &controlpb.CreateFolderRequest{
+		Parent:   "projects/_/buckets/" + b.bucketName,
+		FolderId: folderName,
+	}
+
+	clientFolder, err := b.controlClient.CreateFolder(ctx, req)
+	if err != nil {
+		return nil, err
+	}
+
+	folder := gcs.GCSFolder(b.bucketName, clientFolder)
```

Comment:
refactor  GCSFolder(bucketName string, attrs *controlpb.Folder) signature -->  ConvertToGCSFolder(folerName string, attrs *controlpb.Folder) instead of passing bucket name, we should pass folder name from outside, as bucketname is not being used for any other logic inside method. Folder method should only have folder related method arguments.

Reply:
As per go/go-style/best-practices#function-name-conventions it says even noun names should be good. i.e. gcsFolder(bucketName, attrs *controlpb.Folder) should be fine.

Reply:
> ConvertToGCSFolder(folerName string, attrs *controlpb.Folder) instead of passing bucket name, we should pass folder name from outside, as bucketname is not being used for any other logic inside method. Folder method should only have folder related method arguments.

The bucketName is currently being used within the method to retrieve the folderName. If we implement your suggestion, we would need to call both methods wherever this functionality is required. However, I don't see passing the bucketName as a negative, as it might be necessary to have the bucket name in order to fetch folder attributes.

Reply:
However, I don't see passing the bucketName as a negative, as it might be necessary to have the bucket name in order to fetch folder attributes. >> lets add bucket name when its actually needed for more than 1 usecase, follow [YAGNI principle](https://www.geeksforgeeks.org/what-is-yagni-principle-you-arent-gonna-need-it/) here

Reply:
As per go/go-style/best-practices#function-name-conventions it says even noun names should be good. i.e. gcsFolder(bucketName, attrs *controlpb.Folder) should be fine. >> as per this style guide if method does something then verb should be added, IIUC this methods actually does the conversion hence the suggestion.